engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -263,22 +277,52 @@ def has_no_regcode?(auth_header)
 
         protected
 
+        # rubocop:disable Metrics/PerceivedComplexity
+        # rubocop:disable Metrics/CyclomaticComplexity
         def scc_activate_product
           logger.info "Activating product #{@product.product_string} to SCC"
-          auth = request.headers['HTTP_AUTHORIZATION']
+          auth = nil
+          auth = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
+          mode = nil
           if @system.byos?
-            response = SccProxy.scc_activate_product(@product, auth, params[:token], params[:email])
+            mode = 'byos'
+          elsif !@product.free? && @product.extension?
+            # system is not BYOS and
+            # product is a non free extension
+            mode = 'hybrid'
+            # the extensions must be the same version and arch
+            # than base product
+            base_prod = @system.products.find_by(product_type: :base)
+            if base_prod.present? && @product.arch == base_prod.arch && @product.version == base_prod.version
+              request.headers['proxy_byos_mode'] = mode
+              request.headers['scc_login'] = @system.login
+              request.headers['scc_password'] = @system.password
+              request.headers['instance_data'] = params[:instance_data]
+              request.headers['hwinfo'] = params[:hwinfo]
+              response = SccProxy.announce_system_scc(auth, request.headers)
+            end
+          end
+
+          unless mode.nil?
+            response = SccProxy.scc_activate_product(@product, auth, params, mode)
             unless response.code_type == Net::HTTPCreated
               error = JSON.parse(response.body)
               logger.info "Could not activate #{@product.product_string}, error: #{error['error']} #{response.code}"
               error['error'] = SccProxy.parse_error(error['error']) if error['error'].include? 'json'
               raise ActionController::TranslatedError.new(error['error'])
             end
+            # if the system is PAYG and the registration code is valid for the extension,
+            # then the system is hybrid
+            # update the system to HYBRID mode if HYBRID MODE and system not HYBRID already
+            @system.hybrid! if mode == 'hybrid' && @system.payg?
```

Comment:
How would we end up with `mode == 'hybrid'` and **not** `@system.payg?` ?

Reply:
when a `hybrid` system activates an extension with a registration code ?

Reply:
So, this is in anticipation of a 2nd extension being activated?

Reply:
> when a `hybrid` system activates an extension with a registration code ?

A system can only be hybrid when it is PAYG and an add-on product gets activated. The activation of an add-on product requires a registration code.